### PR TITLE
Sign-up

### PR DESCRIPTION
**Pull request by thanhtt**

### CODE REVIEW

File: Gemfile [hunk 1]
```diff
@@ -1,6 +1,7 @@
 source "https://rubygems.org"
 
 gem "rails", "4.2.6"
+gem 'bcrypt', "3.1.7"
```
Comment (vuducluan) [inline Gemfile]: using "", not ''

File: app/controllers/users_controller.rb [hunk 2]
```diff
@@ -0,0 +1,26 @@
+class UsersController < ApplicationController
+
+  def show
+  	@user = User.find(params[:id])
```
Comment (vuducluan) [inline app/controllers/users_controller.rb]: ident, remove ()

File: Gemfile (hunk 1 above)
Comment (vietmt) [inline Gemfile]: Use double quote

File: app/controllers/users_controller.rb (hunk 2 above)
Comment (vietmt) [inline app/controllers/users_controller.rb]: indent

File: app/controllers/users_controller.rb [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+  end
+
+  def new
+  	@user = User.new
```
Comment (vietmt) [inline app/controllers/users_controller.rb]: indent

File: app/controllers/users_controller.rb [hunk 4]
```diff
@@ -0,0 +1,26 @@
...
+  	if @user.save
+  		flash[:success] = t"welcome"
+  		redirect_to @user
+  	else
+  		render "new"
+  	end
+  end
+
+  private
+
```
Comment (vietmt) [inline app/controllers/users_controller.rb]: Remove this line

File: app/controllers/users_controller.rb (hunk 2 above)
Comment (vietmt) [inline app/controllers/users_controller.rb]: Remove ()

File: app/views/shared/_error_messages.html.erb [hunk 5]
```diff
@@ -0,0 +1,12 @@
+<% if @user.errors.any? %>
+  <div id="error_explanation">
+    <div class="alert alert-danger">
+      The form contains <%= pluralize(@user.errors.count, "error") %>.
```
Comment (vietmt) [inline app/views/shared/_error_messages.html.erb]: I18n

File: app/models/user.rb [hunk 6]
```diff
@@ -0,0 +1,10 @@
+class User < ActiveRecord::Base
+  before_save {self.email = email.downcase}
+  validates :name,  presence: true, length: {maximum: 50}
+  VALID_EMAIL_REGEX = /\A[\w+\-.]+@[a-z\d\-.]+\.[a-z]+\z/i
+  validates :email, presence: true, length: {maximum: 255},
+                    format: {with: VALID_EMAIL_REGEX},
```
Comment (vietmt) [inline app/models/user.rb]: Coding convention

File: Gemfile [hunk 7]
```diff
@@ -16,3 +17,7 @@ group :development do
   gem "byebug"
   gem "sqlite3"
 end
+
+group :production do
+  gem "puma", "3.1.0"
```
Comment (dieunb) [inline Gemfile]: Cần gì cái này?

File: app/views/layouts/application.html.erb [hunk 8]
```diff
@@ -2,14 +2,19 @@
 <html>
   <head>
     <title><%= full_title yield :title %></title>
-    <%= stylesheet_link_tag "application", media: "all", "data-turbolinks-track": true %>
-    <%= javascript_include_tag "application", "data-turbolinks-track": true %>
+    <%= stylesheet_link_tag "application", media: "all",
+      "data-turbolinks-track" => true %>
```
Comment (dieunb) [inline app/views/layouts/application.html.erb]: ` "data-turbolinks-track": true`

File: app/views/layouts/application.html.erb [hunk 9]
```diff
@@ -2,14 +2,19 @@
...
+    <%= javascript_include_tag "application",
+      "data-turbolinks-track" => true %>
```
Comment (dieunb) [inline app/views/layouts/application.html.erb]: ` "data-turbolinks-track": true`

File: config/puma.rb [hunk 10]
```diff
@@ -0,0 +1,13 @@
+workers Integer ENV["WEB_CONCURRENCY"] || 2
```
Comment (dieunb) [inline config/puma.rb]: bỏ đi, ko cần

**Pull request merged**